psi4/src/psi4/libfock/DiskJK.cc

```diff
@@ -75,8 +75,8 @@ void DiskJK::preiterations() {
 
     std::shared_ptr<SOBasisSet> bas = mints->sobasisset();
 
-    so2symblk_ = new int[primary_->nbf()];
-    so2index_ = new int[primary_->nbf()];
+    so2symblk_ = new int[primary_->nbf()]; // lgtm [cpp/resource-not-released-in-destructor]
+    so2index_ = new int[primary_->nbf()]; // lgtm [cpp/resource-not-released-in-destructor]
```

Comment:
these are freed in `postiterations()`. would that be a double free if called from the destructor to be sure?

Reply:
If the method is called in both, it would be a double-free based on how I can tell the code is written.

Having looked into some of the code that calls these methods, `postiterations()` is a function belonging to the jk object, overridden by each class, that gets called at the end of each scf calculation. We could move the `delete` calls to the destructor to appease, but my guess is that whoever wrote this code did it to match the other methods more closely. 

As far as I can tell, someone with more knowledge than I have should decide whether we want to move the `delete` calls to the destructor to appease LGTM or leave it as is, trusting those who wrote this code to have done it best.

Reply:
Do we want to move the delete or keep it in post-iterations?

Reply:
@loriab I'm summoning you because I don't expect you to be checking this